### PR TITLE
docs: Add machine config docs

### PR DESCRIPTION
**Pull request by bradbeam**
Signed-off-by: Brad Beam <brad.beam@talos-systems.com>

### CODE REVIEW

File: docs/content/configuration/v0-reference.md [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
 
 Talos User Data is responsible for the host and Kubernetes configuration, and it is independent of other cloud config / cloud init data.
 
 ## Version
 
 ``Version`` represents the Talos userdata configuration version.
-This denotes
-what the schema of the configuration file.
+
+This denotes what the schema of the configuration file.
```
Comment (andrewrynhard) [inline docs/content/configuration/v0-reference.md]:
```suggestion
This denotes what the schema of the configuration file is.
```

File: docs/content/configuration/v1alpha1-reference.md [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+title: v1alpha1 Reference
+date: 2019-10-04T19:40:55-07:00
+draft: false
+weight: 20
+menu:
+  docs:
+    parent: 'configuration'
+---
+
+Talos User Data is responsible for the host and Kubernetes configuration, and it is independent of other cloud config / cloud init data.
```
Comment (andrewrynhard) [inline docs/content/configuration/v1alpha1-reference.md]:
```suggestion
Talos User Data is responsible for the host and Kubernetes configuration, and it is independent of other cloud init data.
```

Comment (andrewrynhard) [reply docs/content/configuration/v1alpha1-reference.md]: nit: A little unclear to me what it means to be "independent of other cloud config / cloud init data"

File: docs/content/configuration/v1alpha1-reference.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+## Differences from v0
+
+The main driver in introducing a new configuration file format is to reduce the complexity and make it more approachable.
+The redesign proposal can be found [here](https://github.com/talos-systems/talos/blob/master/docs/proposals/20190708-MachineConfig.md).
+The major change between these two versions is the introduction of `machine` and `cluster` configuration objects.
+Machine configuration data deals with the configuration of the host itself whereas cluster configuration data deals with the configuration of the clustering on top of Talos ( ex, Kubernetes configuration ).
```
Comment (andrewrynhard) [inline docs/content/configuration/v1alpha1-reference.md]:
```suggestion
Machine configuration data deals with the configuration of the host itself whereas cluster configuration data deals with the configuration of the cluster on top of Talos ( ex, Kubernetes configuration ).
```

File: docs/content/configuration/v1alpha1-reference.md [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+
+## Version
+
+``Version`` represents the Talos userdata configuration version.
+
+This denotes what the schema of the configuration file.
```
Comment (andrewrynhard) [inline docs/content/configuration/v1alpha1-reference.md]:
```suggestion
This denotes what the schema of the configuration file is.
```

File: docs/content/configuration/v1alpha1-reference.md [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+## Differences from v0
+
+The main driver in introducing a new configuration file format is to reduce the complexity and make it more approachable.
+The redesign proposal can be found [here](https://github.com/talos-systems/talos/blob/master/docs/proposals/20190708-MachineConfig.md).
+The major change between these two versions is the introduction of `machine` and `cluster` configuration objects.
+Machine configuration data deals with the configuration of the host itself whereas cluster configuration data deals with the configuration of the clustering on top of Talos ( ex, Kubernetes configuration ).
+
+## Version
+
+``Version`` represents the Talos userdata configuration version.
```
Comment (andrewrynhard) [inline docs/content/configuration/v1alpha1-reference.md]:
```suggestion
``Version`` represents the Talos configuration version.
```

I'd like to get away from using "user data".

File: docs/content/configuration/v1alpha1-reference.md [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+
+Acceptable values are:
+-``init``
+-``controlplane``
+-``worker``
+
+#### Init
+
+Init node type designates the first control plane node to come up.
+You can think of it like a bootstrap node.
```
Comment (andrewrynhard) [inline docs/content/configuration/v1alpha1-reference.md]: Note for next config version: I think, eventually, it makes send to use `bootstrap` instead if `init`.

File: docs/content/configuration/v1alpha1-reference.md [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+This means it will host etcd along with the Kubernetes master components such as API Server, Controller Manager, Scheduler.
+
+#### Worker
+
+Worker node type designates the node as a worker node.
+This means it will be an available compute node for scheduling workloads.
+
+### machine.token
+
+``token`` is used for authentication to ``trustd`` to confirm the node's identity.
```
Comment (andrewrynhard) [inline docs/content/configuration/v1alpha1-reference.md]: nit: too many place to edit them all in GitHub, but the double "``" isn't required to render as `code`. Can we change this to a single "`"?

Comment (bradbeam) [reply docs/content/configuration/v1alpha1-reference.md]: Was just following previous example; can address later.

**Pull request merged**